### DISCUSSION TITLE
It would be nice to have different log retention periods for different log levels for the same user.

### DISCUSSION THREAD

**Discussion by josegonzalezNorthius** [jongpie/NebulaLogger#689] [Ideas]
Hey,

We recently introduced NebulaLogger in our organization and have integrated it into many different areas. While monitoring an integration, I activated the logs at the info level, but I would like to delete the info level logs earlier than the error level logs. For this reason, I propose the following change. Let me know if this could be a good idea.

Custom Retention by Log Level: Users will be able to define specific retention periods for each log level (e.g., debug, info, warning, error). This ensures that critical logs can be preserved longer while less important logs can be discarded more quickly.

A solution could be to add separate retention fields for each log level instead of a single retention period for all logs.

**Comment by jongpie** [comment]
@josegonzalezNorthius thanks for the suggestion! I think you're right, have granular control over deleting specific `LogEntry__c` would be very nice, especially in orgs that want to retain ERROR/WARN log entries for longer. There are a couple of ways I might implement this, so I'll tinker with a few design ideas to see how to best make this work - I've created issue #698 to track this enhancement.

Let me know if you have any other thoughts/suggestions!
